realm/realm-library/src/objectServer/java/io/realm/ObjectServer.java

```diff
@@ -33,7 +36,7 @@
 @Keep
 class ObjectServer {
 
-    public static void init(Context context) {
+    public static void init(Context context, String appDefinedUserAgent) {
```

Comment:
use an overload or varargs to make this parameter optional, so: 
1- you don't break current API
2- don't force the user to think about this parameter (especially in the context of non sync)

Reply:
I think there is a misunderstanding. This is a briding class only used by internal code (the class is package private). In the public API specifying the user agent is optional